bundleserver/src/test/java/net/discdd/server/BundleTransportToBundleServerTest.java

```diff
@@ -0,0 +1,255 @@
+import net.discdd.grpc.EncryptedBundleId;
+import net.discdd.pathutils.TransportPaths;
+import net.discdd.transport.TransportToBundleServerManager;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.logging.Logger;
+
+import static net.discdd.transport.TransportToBundleServerManager.RECENCY_BLOB_BIN;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+@SpringBootTest(classes = {BundleServerApplication.class, End2EndTest.End2EndTestInitializer.class})
+@TestMethodOrder(MethodOrderer.MethodName.class)
+public class BundleTransportToBundleServerTest extends End2EndTest {
+    private static final Logger logger = Logger.getLogger(BundleTransportToBundleServerTest.class.getName());
+    private static TransportToBundleServerManager manager;
+    private static BundleExchangeServiceGrpc.BundleExchangeServiceStub stub;
+    private static BundleExchangeServiceGrpc.BundleExchangeServiceBlockingStub blockingStub;
+    private static ManagedChannel channel;
+    private static Path toClientPath;
+    private static Path toServerPath;
+
+
+    @BeforeAll
+    static void setUp() throws Exception {
+        // Initialize the TransportPaths class
+        TransportPaths transportPaths = new TransportPaths(End2EndTest.tempRootDir);
+
+        toClientPath = transportPaths.toClientPath;
+        toServerPath = transportPaths.toServerPath;
+
+        manager = new TransportToBundleServerManager(
+                transportPaths,
+                "localhost",
+                Integer.toString(TEST_ADAPTER_GRPC_PORT),
+                (Void) -> {System.out.println("connectComplete"); return null;},
+                (Exception e) -> {System.out.println("connectError"); return null;}
+        );
+    }
+
+    @BeforeEach
+    void setUpEach() {
+        channel = ManagedChannelBuilder.forAddress("127.0.0.1", BUNDLESERVER_GRPC_PORT).usePlaintext().build();
+        stub = BundleExchangeServiceGrpc.newStub(channel);
+        blockingStub = BundleExchangeServiceGrpc.newBlockingStub(channel);
+
+        toClientPath.toFile().mkdirs();
+        toServerPath.toFile().mkdirs();
+    }
+
+    @AfterEach
+    void tearDownEach() throws IOException {
+        // Clean up directories and files
+        if (Files.exists(toClientPath)) {
+            Files.walk(toClientPath)
```

Comment:
you should refactor this into a recursiveDelete method so that you don't duplicate it below.